test/integration/server-selection/readpreference.test.ts

```diff
@@ -463,4 +457,214 @@ describe('ReadPreference', function () {
+        test: async function () {
+          const client = this.configuration.newClient(
+            {
+              readPreference: 'primary',
+              directConnection: true
+            },
+            {
+              monitorCommands: true
+            }
+          );
+          events = [];
+          client.on('commandStarted', event => {
+            if (event.commandName === 'find') {
+              events.push(event);
+            }
+          });
+          let serverError;
+          try {
+            await client.db('test').collection('test').findOne({ a: 1 });
+          } catch (error) {
+            serverError = error;
+          }
+
+          expect(serverError).to.not.exist;
+          expect(events[0]).to.not.containSubset({
+            commandName: 'find',
+            command: {
+              $readPreference: { mode: 'primaryPreferred' }
+            }
+          });
+          expect(events[0]).to.not.containSubset({
+            commandName: 'find',
+            command: {
+              $readPreference: { mode: 'primary' }
+            }
+          });
+
+          await client.close();
+        }
+      });
+    });
+
+    context('when readPreference is secondary', () => {
+      it('should attach a read preference of secondary to the read command for replicaset', {
+        metadata: { requires: { topology: 'replicaset', mongodb: '>=3.6' } },
+        test: async function () {
+          let checkedSecondary = false;
+
+          for (const server of this.configuration.options.hostAddresses) {
+            const { host, port } = server.toHostPort();
+            const client = this.configuration.newClient(
+              {
+                readPreference: 'secondary',
+                directConnection: true,
+                host,
+                port
+              },
+              {
+                monitorCommands: true
+              }
+            );
+            events = [];
+            client.on('commandStarted', event => {
+              if (event.commandName === 'find') {
+                events.push(event);
+              }
+            });
+            let serverStatus;
+
+            try {
+              const admin = client.db().admin();
+              serverStatus = await admin.serverStatus();
+            } catch (serverStatusError) {
+              expect(serverStatusError).to.not.exist;
+            }
+
+            if (serverStatus.repl.secondary) {
+              let serverError;
+
+              try {
+                await client.db('test').collection('test').findOne({ a: 1 });
+              } catch (error) {
+                serverError = error;
+              }
+
+              expect(serverError).to.not.exist;
+              expect(events[0]).to.containSubset({
+                commandName: 'find',
+                command: {
+                  $readPreference: { mode: 'secondary' }
+                }
+              });
```

Comment:
I don't love containSubset it could overlook mistakes that ideally we'd want to be more strict about in our assertions. If I add `events[0].command.$readPreference.funnyMode = 'haha'` this test still passes because it still satisfied the subset, but I think we want to not only make sure readpref is attached but that it is attached correctly. But I don't expect us to make a big manual copy of the command just to check eq, you can get something susinct with has deep property checks:

```ts
expect(events[0]).to.have.property('commandName', 'find');
expect(events[0]).to.have.deep.nested.property('command.$readPrefrence', { mode: 'secondary' })
```